### DISCUSSION TITLE
History behind `DataCollectionOf` attribute

### DISCUSSION THREAD

**Discussion by Mohammad-Alavi** [spatie/laravel-data#659] [Q&A] [answered]
Can anyone expain why [docs](https://spatie.be/docs/laravel-data/v4/as-a-data-transfer-object/nesting) (at the bottom of the page) recommend to use annotation over `DataCollectionOf` attribute?  
I am really curious!

Docs says this about `DataCollectionOf` attribute:
> This was the old way to define the type of data objects that will be stored within a collection. It is still supported, but we recommend using the annotation.

**Comment by rubenvanassche** [comment, marked as answer]
Your IDE will provide a lot more information in completions + tools like PHPStan will be smarter. DataCollectionOf was built at a time support for annotations wasn't as good as it is now and parsing annotations was more difficult.

**Reply by Mohammad-Alavi** [reply]
Thanks!
Just to be sure; so the latest recommendations is to use the annotaion instead of  `DataCollectionOf` attribute despite `DataCollectionOf` being extensively used through the docs?

**Reply by rubenvanassche** [reply]
Well I tried removing the majority of them, but there's probably still some work on that!
